artichoke-backend/src/extn/core/time/mruby.rs

```diff
@@ -130,7 +130,7 @@ unsafe extern "C" fn time_self_at(mrb: *mut sys::mrb_state, _slf: sys::mrb_value
 unsafe extern "C" fn time_self_mkutc(mrb: *mut sys::mrb_state, _slf: sys::mrb_value) -> sys::mrb_value {
     let args = mrb_get_args!(mrb, *args);
     unwrap_interpreter!(mrb, to => guard);
-    let args = args.iter().copied().map(Value::from);
+    let args = &args.iter().copied().map(Value::from).collect::<Vec<Value>>();
```

Comment:
It's tempting to pass this iterator through as it was done previously, however the length of the args is a very quick validation, and makes the rest of the checks a whole lot easier - thus the collection into a Vec. 

However, I think this would be better if we actually just let the `Args` take the `mrb_get_args!` returned slice, and then do all of the operations directly in `Args` itself. The reason, we can avoid the conversion to `Value` fairly quickly by just checking the slice length etc. It could be duplicated logic, but IMO it makes more sense to just use `Args` to parse these values directly. Thoughts?

Reply:
there are a couple of other places in `mruby.rs` files that we allocate a `Vec` for a `*rest` args. I expect this to go away when we are not adapting mruby's value type to Artichoke's, so this seems fine for now.